onchain/wallet/wallet.go

```diff
@@ -642,24 +634,55 @@ func (wallet *Wallet) SearchOutput(txId, address string) (*onchain.Output, error
 	return nil, nil
 }
 
-func (wallet *Wallet) SendToAddress(address string, amount uint64, satPerVbyte float64) (string, error) {
+type outputs struct {
+	Unspent map[string][]map[string]any `json:"unspent_outputs"`
+}
+
+func (wallet *Wallet) getUnspentOutputs(subaccount uint64, includeUnconfirmed bool) (*outputs, error) {
+	details := map[string]any{"subaccount": subaccount}
+	if includeUnconfirmed {
+		details["num_confs"] = 0
+	} else {
+		details["num_confs"] = 1
+	}
+	params, free := toJson(details)
+	defer free()
+
+	handler := new(AuthHandler)
+	result := &outputs{}
+	if err := withAuthHandler(C.GA_get_unspent_outputs(wallet.session, params, handler), handler, result); err != nil {
+		return nil, err
+	}
+	for spent, _ := range wallet.spentOutputs {
+		found := false
+		for key, outputs := range result.Unspent {
+			for i, output := range outputs {
+				if output["txhash"] == spent {
+					result.Unspent[key] = append(outputs[:i], outputs[i+1:]...)
```

Comment:
Wouldn't it be easier to iterate over the outputs of the result and checking if they exist in `wallet.spentOutputs`? It's a map and we should take advantage of that

Reply:
We want to check for the transactions which are in our map, but not in the gdk output set to make sure old ones are deleted, so we have to do it this way